core/src/main/java/org/hibernate/ogm/dialect/impl/GridDialectInitiator.java

```diff
@@ -50,27 +49,24 @@ public GridDialect initiateService(Map configurationValues, ServiceRegistryImple
 		DatastoreProvider datastore = registry.getService( DatastoreProvider.class );
 
 		boolean errorHandlerConfigured = configurationValues.containsKey( OgmProperties.ERROR_HANDLER );
-		EventContextManager eventContext = registry.getService( EventContextManager.class );
 
 		ConfigurationPropertyReader propertyReader = new ConfigurationPropertyReader( configurationValues, registry.getService( ClassLoaderService.class ) );
 
 		return ( (DefaultClassPropertyReaderContext<GridDialect>) propertyReader.property( OgmProperties.GRID_DIALECT, GridDialect.class )
 				.instantiate() )
 				.withDefaultImplementation( registry.getService( DatastoreProvider.class ).getDefaultDialect() )
-				.withInstantiator( new GridDialectInstantiator( datastore, errorHandlerConfigured, eventContext ) )
+				.withInstantiator( new GridDialectInstantiator( datastore, errorHandlerConfigured ) )
 				.getValue();
 	}
 
 	private static class GridDialectInstantiator implements Instantiator<GridDialect> {
 
 		private final DatastoreProvider datastore;
 		private final boolean errorHandlerConfigured;
-		private final EventContextManager eventContext;
 
-		public GridDialectInstantiator(DatastoreProvider datastore, boolean errorHandlerConfigured, EventContextManager eventContext) {
```

Comment:
We can't because there is a dependency between the GridDialect service and the EventContextManager service.

See EventContextManager#isOperationsQueueRequired.